test/src/main/java/org/apache/accumulo/test/BrokenBalancerIT.java

```diff
@@ -0,0 +1,150 @@
+
+  public static class BrokenBalancer extends SimpleLoadBalancer {
+    public BrokenBalancer() {
+      super();
+    }
+
+    public BrokenBalancer(TableId tableId) {
+      super(tableId);
+    }
+
+    @Override
+    public void init(BalancerEnvironment balancerEnvironment) {
+      throw new IllegalStateException();
+    }
+  }
+
+  @Override
+  public void configure(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    Map<String,String> siteConfig = cfg.getSiteConfig();
+    siteConfig.put(Property.TSERV_MAXMEM.getKey(), "10K");
+    siteConfig.put(Property.TSERV_MAJC_DELAY.getKey(), "50ms");
+    siteConfig.put(Property.MANAGER_TABLET_GROUP_WATCHER_INTERVAL.getKey(), "3s");
+    cfg.setSiteConfig(siteConfig);
+    // ensure we have two tservers
+    if (cfg.getNumTservers() != 2) {
+      cfg.setNumTservers(2);
+    }
+  }
+
+  @Test
+  public void testBalancerException() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    testBadBalancer(BrokenBalancer.class.getName(), tableName);
+  }
+
+  @Test
+  public void testBalancerNotFound() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    testBadBalancer("org.apache.accumulo.abc.NonExistentBalancer", tableName);
+  }
+
+  private void testBadBalancer(String balancerClass, String tableName) throws Exception {
+    var executor = Executors.newCachedThreadPool();
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProperties()).build()) {
+      SortedSet<Text> splits = new TreeSet<>();
+      for (int i = 0; i < 10; i++) {
+        splits.add(new Text("" + i));
+      }
+      var props = Map.of(Property.TABLE_LOAD_BALANCER.getKey(), balancerClass);
+      NewTableConfiguration ntc =
+          new NewTableConfiguration().withSplits(splits).setProperties(props);
+      c.tableOperations().create(tableName, ntc);
+
+      var scanFuture = executor.submit(() -> {
+        try (var scanner = c.createScanner(tableName)) {
+          scanner.forEach((k, v) -> System.out.println(k + " " + v));
+        }
+        return 0;
+      });
+
+      UtilWaitThread.sleep(5000);
+      // scan should not be able to complete because the tablet should not be assigned
+      Assertions.assertFalse(scanFuture.isDone());
+
+      // fix the balancer config
+      log.info("fixing per tablet balancer");
+      c.tableOperations().setProperty(tableName, Property.TABLE_LOAD_BALANCER.getKey(),
+          SimpleLoadBalancer.class.getName());
+
+      scanFuture.get();
+
+      // break the balancer at the system level
+      log.info("breaking manager balancer");
+      c.instanceOperations().setProperty(Property.MANAGER_TABLET_BALANCER.getKey(), balancerClass);
+
+      // add some tablet servers
+      assertEquals(2, getCluster().getConfig().getNumTservers());
+      getCluster().getConfig().setNumTservers(5);
+      getCluster().getClusterControl().start(ServerType.TABLET_SERVER);
+      getCluster().getClusterControl().start(ServerType.TABLET_SERVER);
```

Comment:
Is there a reason for the duplicate line? Looking at the code, calling `start(ServerType)` once might end up starting all of them as it passes Integer.MAX_VALUE for the number of processes to start (seems like a bug).

Reply:
I wanted to start 2 tservers here.   Will the first call to start start 3?  I may have misunderstood this.  Was thinking each line would start a single tserver.

Reply:
`MiniAccumuloClusterControl.start(ServerType)` and `MiniAccumuloClusterControl.startAllServers(ServerType)` call the exact same method. This is likely a bug and `start(ServerType)` should only start one server. Fixing that bug might cause other tests to break. It looks like you can start N servers only by calling `MiniAccumuloClusterControl.start(ServerType, Map.of(), N)`

Reply:
That is good to know, I will adjust the test to match the actual behavior.